symfony/plugins/orangehrmAuthenticationPlugin/Controller/Traits/AdministratorAccessTrait.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Authentication\Controller\Traits;
+
+use LogicException;
+use OrangeHRM\Authentication\Controller\AdminPrivilegeControllerInterface;
+use OrangeHRM\Core\Controller\AbstractVueController;
+use OrangeHRM\Framework\Http\RedirectResponse;
+use OrangeHRM\Framework\Http\Request;
+
+trait AdministratorAccessTrait
+{
+    /**
+     * @param Request $request
+     * @return RedirectResponse
+     */
+    public function redirectToAdministratorAccess(Request $request): RedirectResponse
+    {
+        if (!$this instanceof AdminPrivilegeControllerInterface) {
+            throw new LogicException(
+                'Trait should be used in class that implements ' . AdminPrivilegeControllerInterface::class
+            );
+        }
+        if (!$this instanceof AbstractVueController) {
+            throw new LogicException(
+                'Trait should be used in class that extends ' . AbstractVueController::class
+            );
+        }
+
+        $forwardUrl = $this->getCurrentRequest()->getPathInfo();
+        $backUrl = $request->headers->get('referer');
+        return $this->redirect('auth/adminAccess?forward=' . $forwardUrl . '&back=' . $backUrl);
```

Comment:
shall I change this to `forward` instead since then the query won't appear in the URL. Otherwise the browser URL looks very messy 😅 
```suggestion
        return $this->forward(
            AdministratorAccessController::class . '::handle',
            [],
            ['forward' => $forwardUrl, 'back' => $backUrl]
        );
```